src/main/java/com/google/api/gax/retrying/ExceptionRetryAlgorithm.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.retrying;
+
+/**
+ * An exception retry algorithm is responsible for the following operations:
+ *
+ * <ol>
+ * <li>Accepting or rejecting a task for retry depending on the exception thrown by the previous
+ * attempt.
+ * <li>Creating {@link TimedAttemptSettings} for each subsequent retry attempt.
+ * </ol>
+ *
+ * Implementations of this interface must be thread-safe.
+ */
+public interface ExceptionRetryAlgorithm {
+  /**
+   * Creates a next attempt {@link TimedAttemptSettings}.
+   *
+   * @param prevThrowable exception thrown by the previous attempt
+   * @param prevSettings previous attempt settings
+   * @return next attempt settings or {@code prevSettings}, if the implementing algorithm does not
+   * provide specific settings for the next attempt
+   */
+  TimedAttemptSettings createNextAttempt(
+      Throwable prevThrowable, TimedAttemptSettings prevSettings);
+
+  /**
+   * Returns {@code true} if another attempt should be made, or {@code false} otherwise.
+   *
+   * @param prevThrowable exception thrown by the previous attempt
+   * @return {@code true} if another attempt should be made, or {@code false} otherwise
```

Comment:
Makes sense, but if we want to solve that problem, we should probably apply it across the board, instead of just doing it occasionally (e.g. one person doing it for new methods, no one else doing it).